pkg/platform/runtime/buildexpression/buildexpression.go

```diff
@@ -906,6 +906,31 @@ func (e *BuildExpression) SetDefaultTimestamp() error {
 	return nil
 }
 
+// MaybeSetDefaultTimestamp changes the solve node's "at_time" value to "$at_time" if and only if
+// the current value is the given timestamp.
+// Buildscripts prefer to use variables for at_time and define them outside the buildexpression as
+// the expression's commit time.
+// While modern buildexpressions use variables, older ones bake in the commit time. This function
+// exists primarily to update those older buildexpressions for use in buildscripts.
+func (e *BuildExpression) MaybeSetDefaultTimestamp(ts *strfmt.DateTime) error {
+	if ts == nil {
+		return nil // nothing to compare to
+	}
+	atTimeNode, err := e.getSolveAtTimeValue()
+	if err != nil {
+		return errs.Wrap(err, "Could not get %s node", AtTimeKey)
+	}
+	if strings.HasPrefix(*atTimeNode.Str, "$") {
+		return nil
+	}
+	if atTime, err := strfmt.ParseDateTime(*atTimeNode.Str); err == nil && atTime == *ts {
```

Comment:
I'm probably missing something here but what about the case where `err == nil` and `atTime != *ts`? Do we want to do nothing?

Reply:
Yes, as stated in the doc comment, we only change the timestamp to default if it matches the given one. If it doesn't match, we leave it alone because the user has likely explicitly set that timestamp to be something other than the default at_time.